cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -165,71 +160,110 @@ func (gc *GceCache) updateMigBasename(ref GceRef, basename string) {
 // Attempts to regenerate cache if there is a Mig with matching prefix in migs list.
 // TODO(aleksandra-malinowska): reconsider failing when there's a Mig with
 // matching prefix, but instance doesn't belong to it.
-func (gc *GceCache) GetMigForInstance(instance *GceRef) (Mig, error) {
+func (gc *GceCache) GetMigForInstance(instanceRef GceRef) (Mig, error) {
 	gc.cacheMutex.Lock()
 	defer gc.cacheMutex.Unlock()
 
-	if mig, found := gc.instancesCache[*instance]; found {
-		return mig, nil
+	if migRef, found := gc.instanceRefToMigRef[instanceRef]; found {
+		mig, found := gc.migs[migRef]
```

Comment:
Previously list of migs was obtained here by `for _, mig := range gc.GetMigs()  {`. Can we still use GetMigs() here, to copy under migs lock?

Reply:
Yeah - sure. I did not notice we have two mutexes here.
Side question: is that really needed? Was it driven by actual perfromance problemes - or it was done so "just in case"

Reply:
gc.getMig